src/GrumPHP/Task/Phpcs.php

```diff
@@ -70,21 +77,40 @@ public function run(ContextInterface $context)
         $config = $this->getConfiguration();
 
         $arguments = $this->processBuilder->createArgumentsForCommand('phpcs');
-        $arguments->addOptionalArgument('--standard=%s', $config['standard']);
-        $arguments->addOptionalArgument('--warning-severity=0', !$config['show_warnings']);
-        $arguments->addOptionalArgument('--tab-width=%s', $config['tab_width']);
-        $arguments->addOptionalArgument('--encoding=%s', $config['encoding']);
-        $arguments->addOptionalCommaSeparatedArgument('--sniffs=%s', $config['sniffs']);
-        $arguments->addOptionalCommaSeparatedArgument('--ignore=%s', $config['ignore_patterns']);
+        $arguments = $this->addArgumentsFromConfig($arguments, $config);
+        $arguments->add('--report-full');
+        $arguments->add('--report-json');
         $arguments->addFiles($files);
 
         $process = $this->processBuilder->buildProcess($arguments);
         $process->run();
 
         if (!$process->isSuccessful()) {
-            return TaskResult::createFailed($this, $context, $this->formatter->format($process));
+            $output = $this->formatter->format($process);
+            if (!$output) {
```

Comment:
This part looks a bit strange. Why couldn't the formatter just format the message?